crates/bevy_solari/src/scene/blas.rs

```diff
@@ -105,20 +105,16 @@ pub fn compact_raytracing_blas(
     mut blas_manager: ResMut<BlasManager>,
     render_queue: Res<RenderQueue>,
 ) {
-    let mut first_mesh_processed = None;
+    let mut i = 0;
+    let queue_len = blas_manager.compaction_queue.len();
 
     let mut vertices_compacted = 0;
-    while vertices_compacted < MAX_COMPACTION_VERTICES_PER_FRAME
-        && let Some((mesh, vertex_count, compaction_started)) =
-            blas_manager.compaction_queue.pop_front()
+    while let Some((mesh, vertex_count, compaction_started)) =
```

Comment:
Is doing this first ok? Doesn't that mean if we hit this frame's vertex compaction limit that queue item will be dropped / forgotten? My understanding is that this queue is asset-change driven and preserved across frames.

Reply:
This could explain why things were disappearing in the example scene

Reply:
> Is doing this first ok? Doesn't that mean if we hit this frame's vertex compaction limit that queue item will be dropped / forgotten? My understanding is that this queue is asset-change driven and preserved across frames.

Oooh shoot, yes. Good catch, I need to fix that.

Reply:
The issue with the thumbtacks was orthogonal, fixed in the other PR.